### PR TITLE
Milestone 2: Data processing in Models

### PR DESCRIPTION
**Pull request by RichardChileya**
### In this section, I :
  -  Created at least one user 
  -  Created  4 posts written by one of the users 
  - Created at least 6 posts comments for one of the posts

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  def update_posts_counter
+    User.find(user_id).increment!(:posts_counter)
+  end
+
+  # A method which returns the 5 most recent comments for a given post.
+  def recent_five_comments
+    Comment.where(post_id: id).order(updated_at: :desc).limit(5)
+  end
+end
```
Comment (OLIPLICHE) [inline app/models/post.rb:16]: - Please try to fix the mistake and submit it again for review.
- NB: comment methods depends on the post method. please make sure that all of them work
- 
Capture

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    User.find(user_id).increment!(:posts_counter)
+  end
+
```
Comment (OLIPLICHE) [inline app/models/post.rb]: - I will suggest that you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) -` after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private
- Example:
```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
```
Comment (OLIPLICHE) [inline app/models/comment.rb:5]: - You can use the same suggestion I made in post: `after_save :update_comments_counter`

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
```
Comment (OLIPLICHE) [inline app/models/like.rb:6]: - You can use the same suggestion I made in post: `after_save :update_likes_counter`

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :likes
+  has_many :comments
```
Comment (OLIPLICHE) [inline app/models/user.rb]: # Optional:

**Maybe adding the foreign keys would help. example:**

- has_many :posts, foreign_key: 'author_id'
- has_many :comments, foreign_key: 'author_id'
- has_many :likes, foreign_key: 'author_id'

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @RichardChileya, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @RichardChileya 😊, 

# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
- The Pull Request has a proper description ✔️
- The project has met all the technical aspects you needed for this project. Keep rocking!  ✔️


Cheers and Happy coding!👏👏👏

## Optional suggestions

- [ ] [OPTIONAL] Kindly fix the broken MIT license in README and add your own license

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**